server/static/js/admin/controllers.js

```diff
@@ -533,6 +533,29 @@ app.controller("SubmissionListCtrl", ['$scope', '$stateParams', '$window', 'Sear
     $scope.search = function() {
       $scope.getPage($scope.currentPage)
     }
+    
+    $scope.download_zip = function(query, all, courseId) {
+      filename = 'query_(your email)_(current time).zip'
+      Search.download_zip({
+        query: query,
+        all: all,
+        courseId: courseId
+      }, function() {
+        $window.swal({
+          title: 'Success',
+          text:'Saving submissions to ' + filename +
+          '\n Zip of submissions will be ready in Google Cloud Storage ok_grades_bucket in a few minutes',
+          type: 'success',
+          confirmButtonText: 'View zip',
+          cancelButtonText: 'Not now',
+          showCancelButton: true},
+          function() {
+            $window.location = 'https://console.developers.google.com/storage/browser/ok_grades_bucket/';
```

Comment:
I think I looked into that when I did the score export. As far as I could tell, the above URL requires the user to have read access to the object (https://cloud.google.com/storage/docs/cloud-console#_downloadingdata under Accessing Google Developers Console). By default only project owners, editors and team members have read access. 
It might be worth trying out though!